base/channels.jl

```diff
@@ -51,7 +51,6 @@ function fetch(c::Channel)
 end
 
 function take!(c::Channel)
-    !isopen(c) && !isready(c) && throw(closed_exception())
```

Comment:
Well yes, because that is how I found the bug. But it ought to have an explicit test anyway. Will add one.